tests/jballerina-unit-test/src/test/resources/test-src/statements/arrays/array-fill-test.bal

```diff
@@ -323,6 +323,47 @@ function testFiniteTypeArrayFill() returns DEC[] {
     return ar;
 }
 
+function testXMLSubtypesArrayFill() {
+    string errorMsg = "{ballerina/lang.array}IllegalListInsertion";
+    string errorDetails = "{\"message\":\"array of length 0 cannot be expanded into array of length 2 without" +
+    " filler values\"}";
+
+    xml:Element[] a = [];
+    error? result = trap addXmlElement(a, xml `<foo/>`);
+    assertEquality(result is error, true);
+    error err = <error>result;
+    assertEquality(errorMsg, err.message());
+    assertEquality(errorDetails, err.detail().toString());
+
+    xml:Comment[] b = [];
+    result = trap addXmlElement(b, xml `<!-- this is a comment text -->`);
+    assertEquality(result is error, true);
+    err = <error>result;
+    assertEquality(errorMsg, err.message());
+    assertEquality(errorDetails, err.detail().toString());
+
+    xml:ProcessingInstruction[] c = [];
+    result = trap addXmlElement(c, xml `<?xml-stylesheet href="mystyle.css" type="text/css"?>`);
+    assertEquality(result is error, true);
+    err = <error>result;
+    assertEquality(errorMsg, err.message());
+    assertEquality(errorDetails, err.detail().toString());
+
+    xml:Text[] d = [];
+    result = trap addXmlElement(d, xml `Hello World`);
+    assertEquality(result is (), true);
+    assertEquality("[``,`Hello World`]", d.toString());
+
+    xml[] e = [];
+    result = trap addXmlElement(e, xml `Hello World`);
+    assertEquality(result is (), true);
+    assertEquality("[``,`Hello World`]", e.toString());
+}
+
+function addXmlElement(xml[] arr, xml element) {
```

Comment:
Shall we rename this function to something like `setXmlArraySecondElement`? (Because it seemed like appending an xml element to the end of an array)